src/terminal.c

```diff
@@ -465,6 +465,9 @@ ex_terminal(exarg_T *eap)
 	cmd = skipwhite(p);
     }
 
+    if (cmd == NULL || *cmd == NUL)
+	cmd = p_sh;
```

Comment:
@mgedmin I already figure out it. See https://github.com/vim/vim/issues/1974#issuecomment-322343572